### DISCUSSION TITLE
Best way to implement AWS Lambda with fiber and FX

### DISCUSSION THREAD

**Discussion by lucasoares** [uber-go/fx#1107] [Q&A]
Hello.

I have an application that can be run in many different environments (locally, k8s, AWS lambda, gcp cloud functions, etc). All use the same codebase without changing anything in our code, only building the binary with a few different options dependencies.

We also have the possibility to run different domains together in a single binary or split it to run on multiple binaries. Example: I have a user domain and a media domain. I can run both in the same executable or I can run each one in different executables), which guarantees the best performance, smaller binary, etc, making it work for our different scenarios.

We use fiber as our main framework to handle requests, execute middlewares etc. For each deployment kind that can't use only an HTTP server (lambda, for example), we use fiber proxies and adapters. Currently, everything is working without `fx`.

I'm struggling to try to integrate `fx` into our codebase. I think it is the best option for us because of our many different deployment possibilities and it would reduce a lot of boilerplate. For some reason, the lambda is not able to execute our code correctly.

If I may ask: what would be the best configuration/implementation to integrate FX and aws lambda? If using fiber, better.

Thanks!

**Comment by vafrcor** [comment]
@lucasoares : i have same problem while deploying my golang application into AWS Lambda.
when i use uber-fx, the function never shutdown correctly (different from normal go application).
do we need call gracefully shutdown (sigterm) manually?
